### PR TITLE
Allow resetting lazy managers on Symfony 6.2

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Sidekick of https://github.com/symfony/symfony/pull/46752

### CODE REVIEW

**Comment by ostrolucky** [comment]
I've fixed flex issue in 2.7 branch, can you rebase please (some CI matrices will still fail due to unrelated issue with DBAL, but at least some should succeed).

File: Registry.php [hunk 1]
```diff
@@ -75,7 +76,7 @@ private function resetOrClearManager(string $managerName, string $serviceId): vo
 
         assert($manager instanceof EntityManagerInterface);
 
-        if (! $manager instanceof LazyLoadingInterface || $manager->isOpen()) {
+        if ((! $manager instanceof LazyLoadingInterface && ! $manager instanceof LazyGhostObjectInterface) || $manager->isOpen()) {
```
Comment (ostrolucky) [inline Registry.php:79]: feel free to suppress this by adding the LazyGhostObjectInterface here https://github.com/doctrine/DoctrineBundle/blob/2d8779dbf8035bc13ae88b7c8537cfd662a4d689/psalm.xml.dist#L43-L43

Comment (nicolas-grekas) [reply Registry.php:79]: updated, thanks for the hint

**Comment by nicolas-grekas** [comment]
Looks like I need to have a closer look at failures.

**Comment by nicolas-grekas** [comment]
Green after https://github.com/symfony/symfony/pull/46929
Help wanted for the psalm failure.
I'm on vacation starting today, feel free to push on my fork if needed.
You might need to borrow (part of) the patch to fix the builds on lower maintained branches.

**Pull request merged**